yapapi/services.py

```diff
@@ -191,26 +204,159 @@ def _respond_nowait(self, message: Optional[Any], response_to: Optional[ServiceS
     async def get_payload() -> Optional[Payload]:
         """Return the payload (runtime) definition for this service.
 
+        To be overridden by the author of a specific Service class.
+
         If `get_payload` is not implemented, the payload will need to be provided in the
         `Golem.run_service` call.
         """
         pass
 
-    async def start(self):
-        """Implement the `starting` state of the service."""
+    async def start(self) -> AsyncGenerator[WorkItem, Awaitable[List[events.CommandEvent]]]:
+        """Implement the handler for the `starting` state of the service.
+
+        To be overridden by the author of a specific Service class.
+
+        Should perform the minimum set of operations after which the instance of a service can be
+        treated as "started", or, in other words, ready to receive service requests. It's up to the
+        developer of the specific Service class to decide what exact operations constitute a
+        service startup.
+
+        As a handler implementing the [work generator pattern](https://handbook.golem.network/requestor-tutorials/golem-application-fundamentals/hl-api-work-generator-pattern),
+        it's expected to be a generator that yields `WorkItems` (generated using the service's
+        instance of the work context) that are then dispatched to the activity by the engine.
+
+        Results of those batches can then be retrieved by awaiting the values captured from yield
+        statements.
+
+        A clean exit from a handler function triggers the engine to transition the state of the
+        instance to the next stage in service's lifecycle - in this case, to `running`.
+
+        On the other hand, any unhandled exception will cause the instance to be either retried on
+        another provider node, if the Cluster's `respawn_unstarted_instances` argument is set to
+        `True`, which is also the default behavior, or altogether terminated, if
+        `respawn_unstarted_instances` is set to `False`.
+
+        ### Example
+
+        ```
+        async def start(self):
+            # deploy the exe-unit
+            self._ctx.deploy()
+            # start the exe-unit's container
+            self._ctx.start()
```

Comment:
nah ... actually, because deploy/start are implicit unless stated otherwise, you could just omit those two altogether... but it's an example .... you might like to do something else ... again, depending on the runtime